curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-intermediate-oop-by-building-a-platformer-game/64cb34c01b3d856a9a59261d.md

```diff
@@ -0,0 +1,405 @@
+---
+id: 64cb34c01b3d856a9a59261d
+title: Step 96
+challengeType: 0
+dashedName: step-96
+---
+
+# --description--
+
+The next step is to create a list of new `checkpoint` instances using the `CheckPoint` class. 
+
+Start by creating a new `const` variable called `checkpoints` and assign it `checkpointPositions.map()`.
+
+For the map callback function, pass in `checkpoint` for the parameter and implicitly return the creation of a new `CheckPoint` instance with the `checkpoint.x` and `checkpoint.y` values passed in as arguments.
+
+# --hints--
+
+You should assign `checkpointPositions.map()` to a `checkpoints` variable.
+
+```js
+assert.match(code, /\s*const\s+checkpoints\s*=\s*checkpointPositions\.map\s*\([^)]*\)\s*;?/);
+
+```
+
+Your callback function should have a `checkpoint` parameter.
+
+```js
+assert.match(code, /\s*const\s+checkpoints\s*=\s*checkpointPositions\.map\s*\(\s*checkpoint\s*=>\s*[^)]*\s*\)\s*;?/);
```

Comment:
```suggestion
assert.match(code, /\s*const\s+checkpoints\s*=\s*checkpointPositions\.map\s*\(\s*\(?checkpoint\)?\s*=>\s*[^)]*\s*\)\s*;?/);
```
This applies to other steps too. There's a bit inconsistencies with the allowing parameter of the arrow function to be within parenthesis.